### PR TITLE
resolved: two dnssec retry/downgrade tweaks

### PR DESCRIPTION
**Pull request by poettering**
Split out of #17535

### CODE REVIEW

File: src/resolve/resolved-dns-server.c [hunk 1]
```diff
@@ -959,14 +985,16 @@ void dns_server_dump(DnsServer *s, FILE *f) {
                 "\tSeen truncated packet: %s\n"
                 "\tSeen OPT RR getting lost: %s\n"
                 "\tSeen RRSIG RR missing: %s\n"
-                "\tSeen invalid packet: %s\n",
+                "\tSeen invalid packet: %s\n"
+                "\tSeen DO flag not copied: %s\n",
```
Comment (bluca) [inline src/resolve/resolved-dns-server.c]: This message doesn't read very well to me - what about: "DO flag received: yes/no"

Comment (keszybz) [reply src/resolve/resolved-dns-server.c]: "DO flag received" is not clear enough: we don't know *why* it wasn't copied. Maybe "Server dropped DO flag"?

**Comment by keszybz** [comment]
LGTM too.

**Comment by poettering** [comment]
made the suggested message change and rebased. upgrading green label.

**Pull request merged**